File: codigo/1D/EF_barra_2_nodos/deduccion_func_forma_EF_barra_2_nodos.py
# Deducción de las funciones de forma del elemento de barra de 2 nodos

import sympy as sp
from sympy.matrices import Matrix

# Defino las variables simbolicas
u1, u2, x, x1, x2, a1, a0 = sp.symbols('u1 u2 x x1 x2 a1 a0')

r = sp.solve((sp.Eq(u1, a1*x1 + a0),
              sp.Eq(u2, a1*x2 + a0)), (a0, a1))

print('a0 = '); sp.pprint(r[a0])
print('a1 = '); sp.pprint(r[a1])

u = sp.expand(r[a1]*x + r[a0]) # Se define ahora u(x) ya que conocemos a1 y a0
u = sp.collect(u, (u1,u2))     # Se factoriza u2
print('u = '); sp.pprint(u)    # Observe aquí las funciones de forma

File: codigo/1D/EF_barra_2_nodos/deduccion_func_forma_EF_barra_2_nodos.m
%% Deduccion de las funciones de forma del elemento de barra de 2 nodos

syms u1 u2 x x1 x2 a1 a0;
r = solve(u1 == a1*x1 + a0, ...
          u2 == a1*x2 + a0,     a0, a1);
disp('a0 = '); pretty(r.a0)
disp('a1 = '); pretty(r.a1)

u = r.a1*x + r.a0;        % Se define ahora u(x) ya que conocemos a1 y a0
u = collect(u, [u1, u2]); % Se factoriza u1 y y2
disp('u = '); pretty(u)   % Observe aqui las funciones de forma